### ISSUE TITLE
VODS API ​/v2​/vods​/list​/{offset}​/{size} Response Capped at 50 items 

### ISSUE THREAD

**Issue by seifsg** [ant-media/Ant-Media-Server#2945]
### Short description

 A GET request on ​/v2​/vods​/list​/{offset}​/{size} is supposed to display all vods within requested size and offset. In the documentation there is no mention of a size limit. However, in our case the response is always capped at 50 and even if we change the offset the result isn't ordered. Still the videos can be viewed and can be managed via 'Management of Ant Media Server' interface, which indicates a bug in this endpoint.

/v2/vods/count returns that there are 97 videos which is true
/v2​/vods​/list​/0/100 returns only 50

### Environment

* Operating system and version: Ubuntu 18.04.5 LTS
* Java version: openjdk 11.0.10 2021-01-19
* Ant Media Server version: Enterprise Edition 2.2.1 20201029_2042
* Browser name and version: Requests made using NodeJS 14 or Postman to API  Endpoint

### Steps to reproduce
1. Get request on ​/v2​/vods​/list​/{offset}​/{size} endpoint, example ​/v2​/vods​/list​/0/1000

### Expected behavior
1. Get request on ​/v2​/vods​/list​/{offset}​/{size} endpoint, example ​/v2​/vods​/list​/0/1000
2. Response is an array of vod details that is equal or less than the given size

### Actual behavior
1. Get request on ​/v2​/vods​/list​/{offset}​/{size} endpoint, example ​/v2​/vods​/list​/0/1000
2. Response is an array of vod details capped at 50 items or less
 

### Logs
https://pastebin.com/bfLV8i8V

**Comment by mekya** [comment]
Hi @seifsg ,

This is an expected behaviour. We limit the max batch size by 50 in order to not run slow queries in the database

You can get the results by changing the result. I mean you can get page by page

 ​/v2​/vods​/list​/0/50 -> returns items between 0 - 50
 ​/v2​/vods​/list​/50/50 -> returns items between 50 - 100
 ​/v2​/vods​/list​/100/50  -> returns items between 100 - 150

Does it make sense?

Regards,
A. Oguz

**Comment by seifsg** [comment]
Hi @mekya,

Thanks for explaining. I noticed that it was capped and I mentioned in the issue. I had to post here because there are no mentions of this in the documentation. I strongly recommend a bit of improvement to it.

Also I can't go page by page as you say there. Because when I get the response it is not ordered by date by default and that's extra to deal with.

Using the response format, to sort I just add field to sort as the documentation says, so the field is **creationDate** but unfortunately, this only returns a 500 error and that's because that's not what the request accepts.

Again this could be a lot easier if the documentation was a bit clearer. I got it working but only after finding this previous issue that is actually still open: https://github.com/ant-media/Ant-Media-Server/issues/1835

If using the documentation alone isn't enough and I have to look for information within responses of issues then I think I'm not the only one that will waste their time with similar problems.

Can you please consider updating documentation?

Thanks

**Comment by mekya** [comment]
Hi @seifsg ,

Thank you for the feedback. 

Is it ok if we update this doc?

https://antmedia.io/rest/#/VoD%20Rest%20Service/getVodList

**Comment by seifsg** [comment]
@mekya Yes please. That will be very helpful. Thank you very much.

**Comment by mekya** [comment]
Yes we have updated the doc in our end and REST API doc will be updated soon.
